llama_index/vector_stores/metal.py

```diff
@@ -0,0 +1,100 @@
+import math
+from typing import Any, Dict, List, Optional
+from gpt_index.data_structs.node_v2 import Node
+from gpt_index.vector_stores.types import VectorStore, NodeEmbeddingResult, VectorStoreQuery, VectorStoreQueryResult
+
+
+class MetalVectorStore(VectorStore):
+    def __init__(self, api_key: str, client_id: str, index_id: str, filters: Optional[Dict[str, Any]] = None):
+        """Init params."""
+        import_err_msg = (
+            "`metal_sdk` package not found, please run `pip install metal_sdk`"
+        )
+        try:
+            import metal_sdk  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+        from metal_sdk.metal import Metal   # noqa: F401
+
+
+        self.api_key = api_key
+        self.client_id = client_id
+        self.index_id = index_id
+        self.filters = filters
+
+        self.metal_client = Metal(api_key, client_id, index_id)
+        self.stores_text = True
+        self.is_embedding_query = True
+
+    def query(self, query: VectorStoreQuery) -> VectorStoreQueryResult:
+        payload = {
+            "embedding": query.query_embedding,  # Query Embedding
+            "filters": self.filters,  # Metadata Filters
+        }
+        response = self.metal_client.search(payload, limit=query.similarity_top_k)
+
+        nodes = []
+        ids = []
+        similarities = []
+
+        for item in response["data"]:
+            text = item["text"]
+            extra_info = item["metadata"]
+            doc_id = item["metadata"]["doc_id"]
+            id = item["id"]
+            node = Node(text=text, extra_info=extra_info, doc_id=doc_id, id=id)
```

Comment:
should `doc_id` be assigned to the doc source id or the id of the embedding doc? i may be doing this incorrectly rn!

Reply:
hey, apologies for the confusion. The correct thing to do right now (believe it or not) is:
```
            text = item["text"]
            extra_info = item["metadata"]
            doc_id = item["metadata"]["doc_id"]
            id = item["id"]
            node = Node(text=text, extra_info=extra_info, doc_id=id, relationships={DocumentRelationship.SOURCE: doc_id})
```

Reply:
I get confused just typing that out lol. Gonna clean this up soon.

Reply:
ah thank you!